Source/WebKit/UIProcess/ProcessThrottler.cpp

```diff
@@ -317,4 +317,37 @@ void ProcessThrottler::TimedActivity::updateTimer()
         m_timer.startOneShot(m_timeout);
 }
 
+template <typename T>
+static Vector<String> activityNames(const T& activities)
+{
+    Vector<String> names;
+    for (const auto* activity : activities) {
+        if (activity)
+            names.append(activity->name());
+    }
+    return names;
+}
+
+WTF::TextStream& operator<<(WTF::TextStream& ts, const ProcessThrottler& throttler)
+{
+    bool didAppend = false;
+
+    if (auto foregroundActivityNames = activityNames(throttler.m_foregroundActivities); foregroundActivityNames.size()) {
```

Comment:
Actually, seeing how activityNames() is used, it seems pretty wasteful to allocate a brand new vector. I don't think we should create a new vector just for logging, there is clearly no container mutation going on here.